runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -25,38 +25,124 @@
 package net.runelite.client.plugins.clanchat;
 
 import com.google.common.eventbus.Subscribe;
-import java.time.temporal.ChronoUnit;
+import com.google.inject.Provides;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
 import javax.inject.Inject;
+import joptsimple.internal.Strings;
 import net.runelite.api.ChatMessageType;
+import net.runelite.api.ChatModeType;
 import net.runelite.api.ClanMemberRank;
 import net.runelite.api.Client;
 import net.runelite.api.GameState;
+import net.runelite.api.VarClientStr;
+import net.runelite.api.events.GameTick;
 import net.runelite.api.events.SetMessage;
+import net.runelite.api.events.VarClientStrChanged;
 import net.runelite.api.widgets.Widget;
 import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.chat.ChatboxInputListener;
+import net.runelite.client.chat.CommandManager;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.events.ChatboxInput;
+import net.runelite.client.events.PrivateMessageInput;
 import net.runelite.client.game.ClanManager;
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
-import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.overlay.OverlayManager;
 
 @PluginDescriptor(
 	name = "Clan Chat",
-	description = "Add rank icons to users talking in clan chat",
-	tags = {"icons", "rank"}
+	description = "Configuration for Clan Chat plugin.",
+	tags = {"icons", "rank", "chat", "history"}
 )
-public class ClanChatPlugin extends Plugin
+public class ClanChatPlugin extends Plugin implements ChatboxInputListener
 {
+	private static final int MAX_RECENT_CHATS = 10;
+
 	@Inject
 	private Client client;
 
 	@Inject
 	private ClanManager clanManager;
 
-	@Schedule(
-		period = 600,
-		unit = ChronoUnit.MILLIS
-	)
-	public void updateClanChatTitle()
+	@Inject
+	private CommandManager commandManager;
+
+	@Inject
+	private ScheduledExecutorService executor;
+
+	@Inject
+	private ClanChatConfig config;
+
+	@Inject
+	private ClanChatOverlay overlay;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	private List<String> chats = new ArrayList<>();
```

Comment:
Looks like you dont need this property as you are sending it directly to overlay

Reply:
Need it for adding to the list. Gets passed by reference so it is set on the overlay once and can be added/popped on the plugin.